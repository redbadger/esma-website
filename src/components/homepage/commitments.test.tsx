import React from "react";
import renderer from "react-test-renderer";
import Commitments from './commitments';
import { useStaticQuery } from "gatsby";

describe("Commitments", () => {
  it("renders correctly", () => {
    useStaticQuery.mockImplementation(() => {
      return testData;
    });
    const tree = renderer
      .create(
        <Commitments />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const testData = {
  images: {
    edges: [
      {
        node: {
          childImageSharp: {
            fluid: {
              base64: "data:image/jpeg;base64,/9j/2wBDABALDA4M...",
              sizes: "(max-width: 800px) 100vw, 800px",
              src: "/static/test/path1/test-image.jpg",
              srcSet: "/static/test/path1/test-image.jpg 200w",
              aspectRatio: 1.5,
              originalName: "best-practice.jpg",
            },
          },
        },
      },
      {
        node: {
          childImageSharp: {
            fluid: {
              base64: "data:image/jpeg;base64,/9j/2wBDABALDA4M...",
              sizes: "(max-width: 800px) 100vw, 800px",
              src: "/static/test/path1/test-image.jpg",
              srcSet: "/static/test/path1/test-image.jpg 200w",
              aspectRatio: 1.5,
              originalName: "incubator-fund.jpg",
            },
          },
        },
      },
      {
        node: {
          childImageSharp: {
            fluid: {
              base64: "data:image/jpeg;base64,/9j/2wBDABALDA4M...",
              sizes: "(max-width: 800px) 100vw, 800px",
              src: "/static/test/path1/test-image.jpg",
              srcSet: "/static/test/path1/test-image.jpg 200w",
              aspectRatio: 1.5,
              originalName: "life-stage-research.png",
            },
          },
        },
      },
    ],
  },
};
