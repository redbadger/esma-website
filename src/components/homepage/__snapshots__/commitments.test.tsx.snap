// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commitments renders correctly 1`] = `
<section
  className="css-8zpf3h-commitmentsStyles"
>
  <h2>
    Our Commitments
  </h2>
  <div
    className="css-q2a813-commonStyles-commitmentsEntryStyles-CommitmentsEntry"
  >
    <div
      className="left-shift"
    >
      <div
        className="homepage-image-wrapper gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "66.66666666666667%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="data:image/jpeg;base64,/9j/2wBDABALDA4M..."
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 800px) 100vw, 800px"
            srcSet="/static/test/path1/test-image.jpg 200w"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 800px) 100vw, 800px"
            src="/static/test/path1/test-image.jpg"
            srcSet="/static/test/path1/test-image.jpg 200w"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/test/path1/test-image.jpg 200w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" srcset=\\"/static/test/path1/test-image.jpg 200w\\" src=\\"/static/test/path1/test-image.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
      <div>
        <h3>
          Life stages research
        </h3>
        <p>
          Create a consolidated and easily accessible appraisal of the existing academic research relating to the issues affecting an individualâ€™s social mobility from cradle to retirement.
        </p>
        <a
          href="/life-stage-research"
        >
          Read research
        </a>
      </div>
    </div>
  </div>
  <div
    className="css-q2a813-commonStyles-commitmentsEntryStyles-CommitmentsEntry"
  >
    <div
      className="right-shift"
    >
      <div>
        <h3>
          Activity Heat Map
        </h3>
        <p>
          Conduct a review of Business, State and Third Sector efforts across the UK to combat the issues identified in Life Stages Research and produce an open source digital heat map to enable users of the ESMA site to identify how they can collaborate with existing provision to maximise impact.
        </p>
        <span
          className="coming-soon"
        >
          Coming September 2021
        </span>
        <a
          href="/activity-map"
        >
          Find out more
        </a>
      </div>
      <test-file-stub />
    </div>
  </div>
  <div
    className="css-q2a813-commonStyles-commitmentsEntryStyles-CommitmentsEntry"
  >
    <div
      className="left-shift"
    >
      <div
        className="homepage-image-wrapper gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "66.66666666666667%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="data:image/jpeg;base64,/9j/2wBDABALDA4M..."
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 800px) 100vw, 800px"
            srcSet="/static/test/path1/test-image.jpg 200w"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 800px) 100vw, 800px"
            src="/static/test/path1/test-image.jpg"
            srcSet="/static/test/path1/test-image.jpg 200w"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/test/path1/test-image.jpg 200w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" srcset=\\"/static/test/path1/test-image.jpg 200w\\" src=\\"/static/test/path1/test-image.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
      <div>
        <h3>
          Best Practice
        </h3>
        <p>
          Conduct an operational appraisal of the organisations and initiatives mapped in the Activity Heat Map and suggest how they could increase operational efficiency through collaboration locally and nationally.
        </p>
        <span
          className="coming-soon"
        >
          Coming September 2021
        </span>
        <a
          href="/operational-review"
        >
          Find out more
        </a>
      </div>
    </div>
  </div>
  <div
    className="css-q2a813-commonStyles-commitmentsEntryStyles-CommitmentsEntry"
  >
    <div
      className="right-shift"
    >
      <div>
        <h3>
          Incubator Fund
        </h3>
        <p>
          We are creating a Social Mobility Incubator fund, to allow radical and impactful ideas to scale quickly across the UK. This can happen through licensing of best practice materials or enabling exemplar initiatives to grow their impact.
        </p>
        <span
          className="coming-soon"
        >
          Coming soon
        </span>
      </div>
      <div
        className="homepage-image-wrapper gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "66.66666666666667%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="data:image/jpeg;base64,/9j/2wBDABALDA4M..."
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 800px) 100vw, 800px"
            srcSet="/static/test/path1/test-image.jpg 200w"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 800px) 100vw, 800px"
            src="/static/test/path1/test-image.jpg"
            srcSet="/static/test/path1/test-image.jpg 200w"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/test/path1/test-image.jpg 200w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" srcset=\\"/static/test/path1/test-image.jpg 200w\\" src=\\"/static/test/path1/test-image.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
</section>
`;
